### PR TITLE
ConfigurationResolver - disallow empty rule name

### PR DESCRIPTION
**Pull request by keradus**
Closing #1713

### CODE REVIEW

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -350,6 +350,10 @@ private function parseRules()
         $rules = array();
 
         foreach (array_map('trim', explode(',', $this->options['rules'])) as $rule) {
+            if ('' === $rule) {
+                throw new InvalidConfigurationException('Empty rule name is not allowed.');
+            }
+
             if ('-' === $rule[0]) {
```
Comment (GrahamCampbell) [inline src/Console/ConfigurationResolver.php]: We might want to be more careful here, since someone could have any number of `-` in front.

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: Then we are throwing exception for non-existing rule.

Comment (GrahamCampbell) [inline src/Console/ConfigurationResolver.php]: It's still a bit confusing though. I think we should be throwing an invalid configuration exception if a dash occurs anywhere in the string apart from the first character.

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: then why `-` and not `^` or any other char?
In the end, "invalid char in rule name" is just a special case of "name of non-existing rule"

Comment (GrahamCampbell) [inline src/Console/ConfigurationResolver.php]: meh, it's just that `---foo` is being treated as `-foo`, and that's not correct

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: `foo` is `{"foo": true}`
`-foo` is `{"foo": false}`
but
`---foo` is `{"--foo": false}`

Comment (GrahamCampbell) [inline src/Console/ConfigurationResolver.php]: not by your code here - you just ltrim it?

Comment (GrahamCampbell) [inline src/Console/ConfigurationResolver.php]:
```
---foo is {"--foo": false}
```

it should be that, but currently it's actually being treated as:

```
---foo is {"foo": false}
```

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: indeed, you are right.
but out of scope.
but changed ;)

**Pull request merged**